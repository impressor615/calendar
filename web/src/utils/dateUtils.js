import moment from 'moment';

export const monthRange = (date) => {
  const startDate = moment(date).startOf('month').startOf('week');
  const endDate = moment(date).endOf('month').endOf('week');
  const diff = endDate.diff(startDate, 'days') + 1;
  return {
    start_date: startDate,
    end_date: endDate,
    items: Array.from({ length: diff }).map((_, index) => {
      const targetDate = moment(startDate).add(index, 'days');
      return {
        year: targetDate.year(),
        month: (targetDate.month() + 1),
        date: targetDate.date(),
        moment: targetDate,
      };
    }),
  }
};

export const weekRange = (date) => {
  const startDate = moment(date).startOf('week');
  const endDate = moment(date).endOf('week');
  const diff = endDate.diff(startDate, 'days') + 1;
  return {
    start_date: startDate,
    end_date: endDate,
    items: Array.from({ length: diff }).map((_, index) => {
      const targetDate = moment(startDate).add(index, 'days');
      return {
        year: targetDate.year(),
        month: (targetDate.month() + 1),
        date: targetDate.date(),
        moment: targetDate,
      };
    }),
  };
};

export const getHourKey = (hour) => {
  let hourKey = hour.toString();
  if (hour <= 9) {
    hourKey = `0${hourKey}`;
  }
  return hourKey;
};
