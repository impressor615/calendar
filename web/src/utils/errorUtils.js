import CONSTANTS from '../constants';

const { ERRORS } = CONSTANTS;
export const getErrorMsg = (error) => {
  const { response } = error.request;
  const errorObj = JSON.parse(response);
  const msg = ERRORS[errorObj.code].ko || ERRORS['401'].ko;
  return msg;
}