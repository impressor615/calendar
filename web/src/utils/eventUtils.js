export const eventBind = (onEvent, props) => event => onEvent(event, props);