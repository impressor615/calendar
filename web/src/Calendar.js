import React, { Component } from 'react';
import PropTypes from 'prop-types';
import moment from 'moment';

import { MonthCalendar, WeekCalendar } from 'components/Calendar';
import { monthRange, weekRange } from 'utils/dateUtils';
import { fetchEvents, updateServerEvent } from 'utils/fetchUtils';
import { getErrorMsg } from 'utils/errorUtils';

import { withContext } from './AppContext';
import CONSTANTS from './constants';


const { NOTIFICATION } = CONSTANTS;
class Calendar extends Component {
  constructor(props) {
    super(props);
    const {
      currentDate,
    } = this.props;

    this.onToggle = this.onToggle.bind(this);
    this.onDragStart = this.onDragStart.bind(this);
    this.onDrop = this.onDrop.bind(this);
    this.state = {
      ...monthRange(currentDate),
    }
  }

  componentDidMount() {
    this.getEvents();
  }

  componentDidUpdate(prevProps) {
    const prevCurrentDate = prevProps.currentDate;
    const prevType = prevProps.type;
    const { currentDate, type } = this.props;
    if (!prevCurrentDate.isSame(currentDate) || prevType !== type) {
      const rangeData = type === 'month' ? monthRange(currentDate) : weekRange(currentDate);
      this.setState({
        ...rangeData,
      }, () => this.getEvents());
    }
  }

  onToggle(e, props) {
    e.stopPropagation();
    const { onToggle } = this.props;
    const newEvent = this.getModalData(props);
    onToggle(newEvent);
  }

  onDragStart(e, props) {
    const { event } = props;
    e.dataTransfer.setData('event_data', JSON.stringify(event));
  }

  onDrop(e, props) {
    const { events, updateEvents, notify, setLoading } = this.props;
    const { type, hour, dateObj } = props;
    const { year, month, date } = dateObj;
    const event = JSON.parse(e.dataTransfer.getData('event_data'));
    const { title, _id, start_date } = event;

    let newStartDate;
    let newEndDate;
    if (type === 'week') {
      newStartDate = moment(`${year}-${month}-${date} ${hour}:00`, 'YYYY-M-D hh:mm');
      newEndDate = moment(newStartDate).add(1, 'hours');
    }

    if (type === 'month') {
      newStartDate = (
        moment(start_date)
          .year(year)
          .month(month - 1)
          .date(date)
      );
      newEndDate = moment(newStartDate).add(1, 'hours');
    }

    const putData = {
      title,
      start_date: newStartDate.toISOString(),
      end_date: newEndDate.toISOString(),
    };
    setLoading();
    updateServerEvent(_id, putData, events)
      .then((newEvents) => {
        setLoading();
        updateEvents(newEvents);
        notify({
          type: 'success',
          message: NOTIFICATION.update,
        });
      }, (error) => {
        setLoading();
        const message = getErrorMsg(error);
        notify({ message });
      });
  }

  getModalData({ type, dateObj, hour }) {
    const newEvent = {};
    if (type === 'month') {
      const currentHour = moment().hour();
      const startDate = moment(dateObj.moment).hours(currentHour);
      newEvent.start_date = startDate;
      newEvent.end_date = moment(startDate).add(1, 'hours');
    }

    if (type === 'week') {
      const startDate = moment(`${dateObj.year}-${dateObj.month}-${dateObj.date} ${hour}:00`, 'YYYY-M-D hh:mm');
      newEvent.start_date = startDate;
      newEvent.end_date = moment(startDate).add(1, 'hours');
    }

    if (type === 'event') {
      const { start_date, end_date, title, _id } = dateObj;
      newEvent._id = _id;
      newEvent.title = title;
      newEvent.start_date = moment(start_date);
      newEvent.end_date = moment(end_date);
    }

    return newEvent;
  }

  getEvents() {
    const { setLoading, notify, updateEvents } = this.props;
    const { start_date, end_date } = this.state;
    setLoading();
    fetchEvents(start_date, end_date)
    .then((result) => {
      setLoading();
      updateEvents(result);
      this.setState({
        calendar: result,
      });
    }, (error) => {
      setLoading();
      const message = getErrorMsg(error);
      notify({ message });
    });
  }

  render() {
    const {
      type,
      events,
      currentDate,
    } = this.props;
    const {
      range,
    } = this.state;
    return (
      <div className="calendar">
        {
          type === 'month' ? (
            <MonthCalendar
              range={range}
              currentDate={currentDate}
              data={events}
              onToggle={this.onToggle}
              onDragStart={this.onDragStart}
              onDrop={this.onDrop}
            /> 
          ) : (
            <WeekCalendar
              range={range}
              currentDate={currentDate}
              data={events}
              onToggle={this.onToggle}
              onDragStart={this.onDragStart}
              onDrop={this.onDrop}
            />
          )
        }
      </div>
    );
  }
};

Calendar.propTypes = {
  type: PropTypes.string.isRequired,
  currentDate : PropTypes.object.isRequired,
  onToggle: PropTypes.func.isRequired,
  setLoading: PropTypes.func.isRequired,
  notify: PropTypes.func.isRequired,
  updateEvents: PropTypes.func.isRequired,
  events: PropTypes.arrayOf(PropTypes.object).isRequired,
};


export default withContext(Calendar);
