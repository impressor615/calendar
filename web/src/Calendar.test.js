import React from 'react';
import moment from 'moment';

import Calendar from './Calendar';
import {
  MonthCalendar,
  WeekCalendar,
  CalendarDay,
  CalendarTime,
  EventBlock,
} from 'components/Calendar';
import { monthRange } from 'utils/dateUtils';


describe('<Calendar />', () => {
  const onToggle = jest.fn();
  const onDragStart = jest.fn();
  const onDrop = jest.fn();
  const currentDate = moment('2018-10-13');
  const mRange = monthRange(currentDate);
  let wrapper;
  const props = {
    type: 'month',
    currentDate,
    onToggle,
    onDragStart,
    onDrop,
    events: [
      {
        title: 'title',
        start_date: moment(currentDate).toISOString(),
        end_date: moment(currentDate).add(1, 'hours').toISOString(),
      },
    ],
  }

  afterEach(() => {
    wrapper.unmount();
  })

  it('should be rendered properly', () => {
    wrapper = mount(<Calendar {...props} />);
    expect(wrapper.find('.calendar')).toHaveLength(1);
    expect(wrapper.find(MonthCalendar)).toHaveLength(1);
    expect(wrapper.find(CalendarDay)).toHaveLength(mRange.range.length);
    expect(wrapper.find(EventBlock)).toHaveLength(1);

    wrapper.setProps({ type: 'week' });
    wrapper.update();
    expect(wrapper.find(WeekCalendar)).toHaveLength(1);
    expect(wrapper.find(CalendarTime)).toHaveLength(24 * 7);
    expect(wrapper.find(EventBlock)).toHaveLength(1);
  });

  it('should be matched with the snapshot', () => {
    wrapper = mount(<Calendar {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});

