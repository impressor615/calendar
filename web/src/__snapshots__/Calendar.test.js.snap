// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Calendar /> should be matched with the snapshot 1`] = `
<Component
  currentDate={"2018-10-12T15:00:00.000Z"}
  events={
    Array [
      Object {
        "end_date": "2018-10-12T16:00:00.000Z",
        "start_date": "2018-10-12T15:00:00.000Z",
        "title": "title",
      },
    ]
  }
  onDragStart={[MockFunction]}
  onDrop={[MockFunction]}
  onToggle={[MockFunction]}
  type="month"
>
  <Calendar
    currentDate={"2018-10-12T15:00:00.000Z"}
    events={
      Array [
        Object {
          "end_date": "2018-10-12T16:00:00.000Z",
          "start_date": "2018-10-12T15:00:00.000Z",
          "title": "title",
        },
      ]
    }
    isLoading={false}
    isOpen={false}
    notify={[Function]}
    onDragStart={[MockFunction]}
    onDrop={[MockFunction]}
    onToggle={[MockFunction]}
    setLoading={[Function]}
    type="month"
    updateEvents={[Function]}
  >
    <div
      className="calendar"
    >
      <MonthCalendar
        currentDate={"2018-10-12T15:00:00.000Z"}
        data={
          Array [
            Object {
              "end_date": "2018-10-12T16:00:00.000Z",
              "start_date": "2018-10-12T15:00:00.000Z",
              "title": "title",
            },
          ]
        }
        onDragStart={[Function]}
        onDrop={[Function]}
        onToggle={[Function]}
        range={
          Array [
            Object {
              "date": 30,
              "moment": "2018-09-29T15:00:00.000Z",
              "month": 9,
              "year": 2018,
            },
            Object {
              "date": 1,
              "moment": "2018-09-30T15:00:00.000Z",
              "month": 10,
              "year": 2018,
            },
            Object {
              "date": 2,
              "moment": "2018-10-01T15:00:00.000Z",
              "month": 10,
              "year": 2018,
            },
            Object {
              "date": 3,
              "moment": "2018-10-02T15:00:00.000Z",
              "month": 10,
              "year": 2018,
            },
            Object {
              "date": 4,
              "moment": "2018-10-03T15:00:00.000Z",
              "month": 10,
              "year": 2018,
            },
            Object {
              "date": 5,
              "moment": "2018-10-04T15:00:00.000Z",
              "month": 10,
              "year": 2018,
            },
            Object {
              "date": 6,
              "moment": "2018-10-05T15:00:00.000Z",
              "month": 10,
              "year": 2018,
            },
            Object {
              "date": 7,
              "moment": "2018-10-06T15:00:00.000Z",
              "month": 10,
              "year": 2018,
            },
            Object {
              "date": 8,
              "moment": "2018-10-07T15:00:00.000Z",
              "month": 10,
              "year": 2018,
            },
            Object {
              "date": 9,
              "moment": "2018-10-08T15:00:00.000Z",
              "month": 10,
              "year": 2018,
            },
            Object {
              "date": 10,
              "moment": "2018-10-09T15:00:00.000Z",
              "month": 10,
              "year": 2018,
            },
            Object {
              "date": 11,
              "moment": "2018-10-10T15:00:00.000Z",
              "month": 10,
              "year": 2018,
            },
            Object {
              "date": 12,
              "moment": "2018-10-11T15:00:00.000Z",
              "month": 10,
              "year": 2018,
            },
            Object {
              "date": 13,
              "moment": "2018-10-12T15:00:00.000Z",
              "month": 10,
              "year": 2018,
            },
            Object {
              "date": 14,
              "moment": "2018-10-13T15:00:00.000Z",
              "month": 10,
              "year": 2018,
            },
            Object {
              "date": 15,
              "moment": "2018-10-14T15:00:00.000Z",
              "month": 10,
              "year": 2018,
            },
            Object {
              "date": 16,
              "moment": "2018-10-15T15:00:00.000Z",
              "month": 10,
              "year": 2018,
            },
            Object {
              "date": 17,
              "moment": "2018-10-16T15:00:00.000Z",
              "month": 10,
              "year": 2018,
            },
            Object {
              "date": 18,
              "moment": "2018-10-17T15:00:00.000Z",
              "month": 10,
              "year": 2018,
            },
            Object {
              "date": 19,
              "moment": "2018-10-18T15:00:00.000Z",
              "month": 10,
              "year": 2018,
            },
            Object {
              "date": 20,
              "moment": "2018-10-19T15:00:00.000Z",
              "month": 10,
              "year": 2018,
            },
            Object {
              "date": 21,
              "moment": "2018-10-20T15:00:00.000Z",
              "month": 10,
              "year": 2018,
            },
            Object {
              "date": 22,
              "moment": "2018-10-21T15:00:00.000Z",
              "month": 10,
              "year": 2018,
            },
            Object {
              "date": 23,
              "moment": "2018-10-22T15:00:00.000Z",
              "month": 10,
              "year": 2018,
            },
            Object {
              "date": 24,
              "moment": "2018-10-23T15:00:00.000Z",
              "month": 10,
              "year": 2018,
            },
            Object {
              "date": 25,
              "moment": "2018-10-24T15:00:00.000Z",
              "month": 10,
              "year": 2018,
            },
            Object {
              "date": 26,
              "moment": "2018-10-25T15:00:00.000Z",
              "month": 10,
              "year": 2018,
            },
            Object {
              "date": 27,
              "moment": "2018-10-26T15:00:00.000Z",
              "month": 10,
              "year": 2018,
            },
            Object {
              "date": 28,
              "moment": "2018-10-27T15:00:00.000Z",
              "month": 10,
              "year": 2018,
            },
            Object {
              "date": 29,
              "moment": "2018-10-28T15:00:00.000Z",
              "month": 10,
              "year": 2018,
            },
            Object {
              "date": 30,
              "moment": "2018-10-29T15:00:00.000Z",
              "month": 10,
              "year": 2018,
            },
            Object {
              "date": 31,
              "moment": "2018-10-30T15:00:00.000Z",
              "month": 10,
              "year": 2018,
            },
            Object {
              "date": 1,
              "moment": "2018-10-31T15:00:00.000Z",
              "month": 11,
              "year": 2018,
            },
            Object {
              "date": 2,
              "moment": "2018-11-01T15:00:00.000Z",
              "month": 11,
              "year": 2018,
            },
            Object {
              "date": 3,
              "moment": "2018-11-02T15:00:00.000Z",
              "month": 11,
              "year": 2018,
            },
          ]
        }
      >
        <section
          className="month-calendar"
        >
          <CalendarDay
            currentDate={"2018-10-12T15:00:00.000Z"}
            dateObj={
              Object {
                "date": 30,
                "moment": "2018-09-29T15:00:00.000Z",
                "month": 9,
                "year": 2018,
              }
            }
            day="일"
            key="0"
            onDrop={[Function]}
            onToggle={[Function]}
          >
            <div
              className="cal-block"
              onClick={[Function]}
              onDragOver={[Function]}
              onDrop={[Function]}
            >
              <div
                className="day"
              >
                일
              </div>
              <div>
                9월 30일
              </div>
            </div>
          </CalendarDay>
          <CalendarDay
            currentDate={"2018-10-12T15:00:00.000Z"}
            dateObj={
              Object {
                "date": 1,
                "moment": "2018-09-30T15:00:00.000Z",
                "month": 10,
                "year": 2018,
              }
            }
            day="월"
            key="1"
            onDrop={[Function]}
            onToggle={[Function]}
          >
            <div
              className="cal-block"
              onClick={[Function]}
              onDragOver={[Function]}
              onDrop={[Function]}
            >
              <div
                className="day"
              >
                월
              </div>
              <div>
                1
              </div>
            </div>
          </CalendarDay>
          <CalendarDay
            currentDate={"2018-10-12T15:00:00.000Z"}
            dateObj={
              Object {
                "date": 2,
                "moment": "2018-10-01T15:00:00.000Z",
                "month": 10,
                "year": 2018,
              }
            }
            day="화"
            key="2"
            onDrop={[Function]}
            onToggle={[Function]}
          >
            <div
              className="cal-block"
              onClick={[Function]}
              onDragOver={[Function]}
              onDrop={[Function]}
            >
              <div
                className="day"
              >
                화
              </div>
              <div>
                2
              </div>
            </div>
          </CalendarDay>
          <CalendarDay
            currentDate={"2018-10-12T15:00:00.000Z"}
            dateObj={
              Object {
                "date": 3,
                "moment": "2018-10-02T15:00:00.000Z",
                "month": 10,
                "year": 2018,
              }
            }
            day="수"
            key="3"
            onDrop={[Function]}
            onToggle={[Function]}
          >
            <div
              className="cal-block"
              onClick={[Function]}
              onDragOver={[Function]}
              onDrop={[Function]}
            >
              <div
                className="day"
              >
                수
              </div>
              <div>
                3
              </div>
            </div>
          </CalendarDay>
          <CalendarDay
            currentDate={"2018-10-12T15:00:00.000Z"}
            dateObj={
              Object {
                "date": 4,
                "moment": "2018-10-03T15:00:00.000Z",
                "month": 10,
                "year": 2018,
              }
            }
            day="목"
            key="4"
            onDrop={[Function]}
            onToggle={[Function]}
          >
            <div
              className="cal-block"
              onClick={[Function]}
              onDragOver={[Function]}
              onDrop={[Function]}
            >
              <div
                className="day"
              >
                목
              </div>
              <div>
                4
              </div>
            </div>
          </CalendarDay>
          <CalendarDay
            currentDate={"2018-10-12T15:00:00.000Z"}
            dateObj={
              Object {
                "date": 5,
                "moment": "2018-10-04T15:00:00.000Z",
                "month": 10,
                "year": 2018,
              }
            }
            day="금"
            key="5"
            onDrop={[Function]}
            onToggle={[Function]}
          >
            <div
              className="cal-block"
              onClick={[Function]}
              onDragOver={[Function]}
              onDrop={[Function]}
            >
              <div
                className="day"
              >
                금
              </div>
              <div>
                5
              </div>
            </div>
          </CalendarDay>
          <CalendarDay
            currentDate={"2018-10-12T15:00:00.000Z"}
            dateObj={
              Object {
                "date": 6,
                "moment": "2018-10-05T15:00:00.000Z",
                "month": 10,
                "year": 2018,
              }
            }
            day="토"
            key="6"
            onDrop={[Function]}
            onToggle={[Function]}
          >
            <div
              className="cal-block"
              onClick={[Function]}
              onDragOver={[Function]}
              onDrop={[Function]}
            >
              <div
                className="day"
              >
                토
              </div>
              <div>
                6
              </div>
            </div>
          </CalendarDay>
          <CalendarDay
            currentDate={"2018-10-12T15:00:00.000Z"}
            dateObj={
              Object {
                "date": 7,
                "moment": "2018-10-06T15:00:00.000Z",
                "month": 10,
                "year": 2018,
              }
            }
            day=""
            key="7"
            onDrop={[Function]}
            onToggle={[Function]}
          >
            <div
              className="cal-block"
              onClick={[Function]}
              onDragOver={[Function]}
              onDrop={[Function]}
            >
              <div>
                7
              </div>
            </div>
          </CalendarDay>
          <CalendarDay
            currentDate={"2018-10-12T15:00:00.000Z"}
            dateObj={
              Object {
                "date": 8,
                "moment": "2018-10-07T15:00:00.000Z",
                "month": 10,
                "year": 2018,
              }
            }
            day=""
            key="8"
            onDrop={[Function]}
            onToggle={[Function]}
          >
            <div
              className="cal-block"
              onClick={[Function]}
              onDragOver={[Function]}
              onDrop={[Function]}
            >
              <div>
                8
              </div>
            </div>
          </CalendarDay>
          <CalendarDay
            currentDate={"2018-10-12T15:00:00.000Z"}
            dateObj={
              Object {
                "date": 9,
                "moment": "2018-10-08T15:00:00.000Z",
                "month": 10,
                "year": 2018,
              }
            }
            day=""
            key="9"
            onDrop={[Function]}
            onToggle={[Function]}
          >
            <div
              className="cal-block"
              onClick={[Function]}
              onDragOver={[Function]}
              onDrop={[Function]}
            >
              <div>
                9
              </div>
            </div>
          </CalendarDay>
          <CalendarDay
            currentDate={"2018-10-12T15:00:00.000Z"}
            dateObj={
              Object {
                "date": 10,
                "moment": "2018-10-09T15:00:00.000Z",
                "month": 10,
                "year": 2018,
              }
            }
            day=""
            key="10"
            onDrop={[Function]}
            onToggle={[Function]}
          >
            <div
              className="cal-block"
              onClick={[Function]}
              onDragOver={[Function]}
              onDrop={[Function]}
            >
              <div>
                10
              </div>
            </div>
          </CalendarDay>
          <CalendarDay
            currentDate={"2018-10-12T15:00:00.000Z"}
            dateObj={
              Object {
                "date": 11,
                "moment": "2018-10-10T15:00:00.000Z",
                "month": 10,
                "year": 2018,
              }
            }
            day=""
            key="11"
            onDrop={[Function]}
            onToggle={[Function]}
          >
            <div
              className="cal-block"
              onClick={[Function]}
              onDragOver={[Function]}
              onDrop={[Function]}
            >
              <div>
                11
              </div>
            </div>
          </CalendarDay>
          <CalendarDay
            currentDate={"2018-10-12T15:00:00.000Z"}
            dateObj={
              Object {
                "date": 12,
                "moment": "2018-10-11T15:00:00.000Z",
                "month": 10,
                "year": 2018,
              }
            }
            day=""
            key="12"
            onDrop={[Function]}
            onToggle={[Function]}
          >
            <div
              className="cal-block"
              onClick={[Function]}
              onDragOver={[Function]}
              onDrop={[Function]}
            >
              <div>
                12
              </div>
            </div>
          </CalendarDay>
          <CalendarDay
            currentDate={"2018-10-12T15:00:00.000Z"}
            dateObj={
              Object {
                "date": 13,
                "moment": "2018-10-12T15:00:00.000Z",
                "month": 10,
                "year": 2018,
              }
            }
            day=""
            key="13"
            onDrop={[Function]}
            onToggle={[Function]}
          >
            <div
              className="cal-block"
              onClick={[Function]}
              onDragOver={[Function]}
              onDrop={[Function]}
            >
              <div>
                13
              </div>
              <EventBlock
                className="primary"
                fluid={false}
                key="0"
                onDragStart={[Function]}
                onToggle={[Function]}
              >
                <Button
                  active={false}
                  className="event-block primary"
                  color=""
                  draggable={true}
                  onClick={[Function]}
                  onDragStart={[Function]}
                >
                  <button
                    className="btn event-block primary"
                    draggable={true}
                    onClick={[Function]}
                    onDragStart={[Function]}
                    type="button"
                  >
                    ○ 
                    00:00 title
                  </button>
                </Button>
              </EventBlock>
            </div>
          </CalendarDay>
          <CalendarDay
            currentDate={"2018-10-12T15:00:00.000Z"}
            dateObj={
              Object {
                "date": 14,
                "moment": "2018-10-13T15:00:00.000Z",
                "month": 10,
                "year": 2018,
              }
            }
            day=""
            key="14"
            onDrop={[Function]}
            onToggle={[Function]}
          >
            <div
              className="cal-block"
              onClick={[Function]}
              onDragOver={[Function]}
              onDrop={[Function]}
            >
              <div>
                14
              </div>
            </div>
          </CalendarDay>
          <CalendarDay
            currentDate={"2018-10-12T15:00:00.000Z"}
            dateObj={
              Object {
                "date": 15,
                "moment": "2018-10-14T15:00:00.000Z",
                "month": 10,
                "year": 2018,
              }
            }
            day=""
            key="15"
            onDrop={[Function]}
            onToggle={[Function]}
          >
            <div
              className="cal-block"
              onClick={[Function]}
              onDragOver={[Function]}
              onDrop={[Function]}
            >
              <div>
                15
              </div>
            </div>
          </CalendarDay>
          <CalendarDay
            currentDate={"2018-10-12T15:00:00.000Z"}
            dateObj={
              Object {
                "date": 16,
                "moment": "2018-10-15T15:00:00.000Z",
                "month": 10,
                "year": 2018,
              }
            }
            day=""
            key="16"
            onDrop={[Function]}
            onToggle={[Function]}
          >
            <div
              className="cal-block"
              onClick={[Function]}
              onDragOver={[Function]}
              onDrop={[Function]}
            >
              <div>
                16
              </div>
            </div>
          </CalendarDay>
          <CalendarDay
            currentDate={"2018-10-12T15:00:00.000Z"}
            dateObj={
              Object {
                "date": 17,
                "moment": "2018-10-16T15:00:00.000Z",
                "month": 10,
                "year": 2018,
              }
            }
            day=""
            key="17"
            onDrop={[Function]}
            onToggle={[Function]}
          >
            <div
              className="cal-block"
              onClick={[Function]}
              onDragOver={[Function]}
              onDrop={[Function]}
            >
              <div>
                17
              </div>
            </div>
          </CalendarDay>
          <CalendarDay
            currentDate={"2018-10-12T15:00:00.000Z"}
            dateObj={
              Object {
                "date": 18,
                "moment": "2018-10-17T15:00:00.000Z",
                "month": 10,
                "year": 2018,
              }
            }
            day=""
            key="18"
            onDrop={[Function]}
            onToggle={[Function]}
          >
            <div
              className="cal-block"
              onClick={[Function]}
              onDragOver={[Function]}
              onDrop={[Function]}
            >
              <div>
                18
              </div>
            </div>
          </CalendarDay>
          <CalendarDay
            currentDate={"2018-10-12T15:00:00.000Z"}
            dateObj={
              Object {
                "date": 19,
                "moment": "2018-10-18T15:00:00.000Z",
                "month": 10,
                "year": 2018,
              }
            }
            day=""
            key="19"
            onDrop={[Function]}
            onToggle={[Function]}
          >
            <div
              className="cal-block"
              onClick={[Function]}
              onDragOver={[Function]}
              onDrop={[Function]}
            >
              <div>
                19
              </div>
            </div>
          </CalendarDay>
          <CalendarDay
            currentDate={"2018-10-12T15:00:00.000Z"}
            dateObj={
              Object {
                "date": 20,
                "moment": "2018-10-19T15:00:00.000Z",
                "month": 10,
                "year": 2018,
              }
            }
            day=""
            key="20"
            onDrop={[Function]}
            onToggle={[Function]}
          >
            <div
              className="cal-block"
              onClick={[Function]}
              onDragOver={[Function]}
              onDrop={[Function]}
            >
              <div>
                20
              </div>
            </div>
          </CalendarDay>
          <CalendarDay
            currentDate={"2018-10-12T15:00:00.000Z"}
            dateObj={
              Object {
                "date": 21,
                "moment": "2018-10-20T15:00:00.000Z",
                "month": 10,
                "year": 2018,
              }
            }
            day=""
            key="21"
            onDrop={[Function]}
            onToggle={[Function]}
          >
            <div
              className="cal-block"
              onClick={[Function]}
              onDragOver={[Function]}
              onDrop={[Function]}
            >
              <div>
                21
              </div>
            </div>
          </CalendarDay>
          <CalendarDay
            currentDate={"2018-10-12T15:00:00.000Z"}
            dateObj={
              Object {
                "date": 22,
                "moment": "2018-10-21T15:00:00.000Z",
                "month": 10,
                "year": 2018,
              }
            }
            day=""
            key="22"
            onDrop={[Function]}
            onToggle={[Function]}
          >
            <div
              className="cal-block"
              onClick={[Function]}
              onDragOver={[Function]}
              onDrop={[Function]}
            >
              <div>
                22
              </div>
            </div>
          </CalendarDay>
          <CalendarDay
            currentDate={"2018-10-12T15:00:00.000Z"}
            dateObj={
              Object {
                "date": 23,
                "moment": "2018-10-22T15:00:00.000Z",
                "month": 10,
                "year": 2018,
              }
            }
            day=""
            key="23"
            onDrop={[Function]}
            onToggle={[Function]}
          >
            <div
              className="cal-block"
              onClick={[Function]}
              onDragOver={[Function]}
              onDrop={[Function]}
            >
              <div>
                23
              </div>
            </div>
          </CalendarDay>
          <CalendarDay
            currentDate={"2018-10-12T15:00:00.000Z"}
            dateObj={
              Object {
                "date": 24,
                "moment": "2018-10-23T15:00:00.000Z",
                "month": 10,
                "year": 2018,
              }
            }
            day=""
            key="24"
            onDrop={[Function]}
            onToggle={[Function]}
          >
            <div
              className="cal-block"
              onClick={[Function]}
              onDragOver={[Function]}
              onDrop={[Function]}
            >
              <div>
                24
              </div>
            </div>
          </CalendarDay>
          <CalendarDay
            currentDate={"2018-10-12T15:00:00.000Z"}
            dateObj={
              Object {
                "date": 25,
                "moment": "2018-10-24T15:00:00.000Z",
                "month": 10,
                "year": 2018,
              }
            }
            day=""
            key="25"
            onDrop={[Function]}
            onToggle={[Function]}
          >
            <div
              className="cal-block"
              onClick={[Function]}
              onDragOver={[Function]}
              onDrop={[Function]}
            >
              <div>
                25
              </div>
            </div>
          </CalendarDay>
          <CalendarDay
            currentDate={"2018-10-12T15:00:00.000Z"}
            dateObj={
              Object {
                "date": 26,
                "moment": "2018-10-25T15:00:00.000Z",
                "month": 10,
                "year": 2018,
              }
            }
            day=""
            key="26"
            onDrop={[Function]}
            onToggle={[Function]}
          >
            <div
              className="cal-block"
              onClick={[Function]}
              onDragOver={[Function]}
              onDrop={[Function]}
            >
              <div>
                26
              </div>
            </div>
          </CalendarDay>
          <CalendarDay
            currentDate={"2018-10-12T15:00:00.000Z"}
            dateObj={
              Object {
                "date": 27,
                "moment": "2018-10-26T15:00:00.000Z",
                "month": 10,
                "year": 2018,
              }
            }
            day=""
            key="27"
            onDrop={[Function]}
            onToggle={[Function]}
          >
            <div
              className="cal-block"
              onClick={[Function]}
              onDragOver={[Function]}
              onDrop={[Function]}
            >
              <div>
                27
              </div>
            </div>
          </CalendarDay>
          <CalendarDay
            currentDate={"2018-10-12T15:00:00.000Z"}
            dateObj={
              Object {
                "date": 28,
                "moment": "2018-10-27T15:00:00.000Z",
                "month": 10,
                "year": 2018,
              }
            }
            day=""
            key="28"
            onDrop={[Function]}
            onToggle={[Function]}
          >
            <div
              className="cal-block"
              onClick={[Function]}
              onDragOver={[Function]}
              onDrop={[Function]}
            >
              <div>
                28
              </div>
            </div>
          </CalendarDay>
          <CalendarDay
            currentDate={"2018-10-12T15:00:00.000Z"}
            dateObj={
              Object {
                "date": 29,
                "moment": "2018-10-28T15:00:00.000Z",
                "month": 10,
                "year": 2018,
              }
            }
            day=""
            key="29"
            onDrop={[Function]}
            onToggle={[Function]}
          >
            <div
              className="cal-block"
              onClick={[Function]}
              onDragOver={[Function]}
              onDrop={[Function]}
            >
              <div>
                29
              </div>
            </div>
          </CalendarDay>
          <CalendarDay
            currentDate={"2018-10-12T15:00:00.000Z"}
            dateObj={
              Object {
                "date": 30,
                "moment": "2018-10-29T15:00:00.000Z",
                "month": 10,
                "year": 2018,
              }
            }
            day=""
            key="30"
            onDrop={[Function]}
            onToggle={[Function]}
          >
            <div
              className="cal-block"
              onClick={[Function]}
              onDragOver={[Function]}
              onDrop={[Function]}
            >
              <div>
                30
              </div>
            </div>
          </CalendarDay>
          <CalendarDay
            currentDate={"2018-10-12T15:00:00.000Z"}
            dateObj={
              Object {
                "date": 31,
                "moment": "2018-10-30T15:00:00.000Z",
                "month": 10,
                "year": 2018,
              }
            }
            day=""
            key="31"
            onDrop={[Function]}
            onToggle={[Function]}
          >
            <div
              className="cal-block"
              onClick={[Function]}
              onDragOver={[Function]}
              onDrop={[Function]}
            >
              <div>
                31
              </div>
            </div>
          </CalendarDay>
          <CalendarDay
            currentDate={"2018-10-12T15:00:00.000Z"}
            dateObj={
              Object {
                "date": 1,
                "moment": "2018-10-31T15:00:00.000Z",
                "month": 11,
                "year": 2018,
              }
            }
            day=""
            key="32"
            onDrop={[Function]}
            onToggle={[Function]}
          >
            <div
              className="cal-block"
              onClick={[Function]}
              onDragOver={[Function]}
              onDrop={[Function]}
            >
              <div>
                11월 1일
              </div>
            </div>
          </CalendarDay>
          <CalendarDay
            currentDate={"2018-10-12T15:00:00.000Z"}
            dateObj={
              Object {
                "date": 2,
                "moment": "2018-11-01T15:00:00.000Z",
                "month": 11,
                "year": 2018,
              }
            }
            day=""
            key="33"
            onDrop={[Function]}
            onToggle={[Function]}
          >
            <div
              className="cal-block"
              onClick={[Function]}
              onDragOver={[Function]}
              onDrop={[Function]}
            >
              <div>
                11월 2일
              </div>
            </div>
          </CalendarDay>
          <CalendarDay
            currentDate={"2018-10-12T15:00:00.000Z"}
            dateObj={
              Object {
                "date": 3,
                "moment": "2018-11-02T15:00:00.000Z",
                "month": 11,
                "year": 2018,
              }
            }
            day=""
            key="34"
            onDrop={[Function]}
            onToggle={[Function]}
          >
            <div
              className="cal-block"
              onClick={[Function]}
              onDragOver={[Function]}
              onDrop={[Function]}
            >
              <div>
                11월 3일
              </div>
            </div>
          </CalendarDay>
        </section>
      </MonthCalendar>
    </div>
  </Calendar>
</Component>
`;
